e2e-tests/msentraid/common/broker.resource

```diff
@@ -0,0 +1,388 @@
+${ENTRAID_BROKER_CFG}    /var/snap/authd-msentraid/current/broker.conf
+${ENTRAID_BROKER_CFG_DIR}    /var/snap/authd-msentraid/current/broker.conf.d
+
+
+*** Keywords ***
+Enable Edge Broker
+    Open GNOME Terminal In Sudo Mode
+    Run Command In GNOME Terminal    snap refresh authd-msentraid --edge
+    Close GNOME Terminal In Sudo Mode
+
+
+Disable Broker Snap
+    Open GNOME Terminal In Sudo Mode
+    Run Command In GNOME Terminal    snap stop authd-msentraid
+    Run Command In GNOME Terminal    rm ${AUTHD_BROKER_CFG}
+    Run Command In GNOME Terminal    systemctl restart authd.service
+    Close GNOME Terminal In Sudo Mode
+
+
+Log In With Remote User Through CLI: QR Code
+    [Arguments]    ${username}    ${domain}    ${local_password}
+    Open GNOME Terminal
+    Start Log In With Remote User Through CLI: QR Code    ${username}    ${domain}
+    Select Provider
+    Continue Log In With Remote User: Log In On Browser    ${username}    ${domain}
+    Continue Log In With Remote User Through CLI: Define Local Password    ${username}    ${domain}    ${local_password}
+
+
+Start Log In With Remote User Through CLI: QR Code
+    [Arguments]    ${username}    ${domain}
+    # TODO: Workaround for the issue that `PlatformHid.Type String` doesnt
+    #       correctly enter special characters which require a Shift modifier
+    PlatformHid.Type String    sudo login ${username}
+    PlatformHid.Keys Combo    Shift_L    @
+    PlatformHid.Type String    ${domain}
+    PlatformHid.Keys Combo    Return
+
+    # Wait for the password prompt
+    PlatformVideoInput.Match Text    [sudo] password for ubuntu
+
+    # Enter the sudo password
+    PlatformHid.Type String    ubuntu
+    PlatformHid.Keys Combo    Return
+
+    # Check that the provider selection contains the Entra ID provider
+    PlatformVideoInput.Match Text    Select your provider    15
+
+
+Select Provider
+    # Check that the provider selection contains the Entra ID provider
+    PlatformVideoInput.Match Text    Select your provider    15
+    PlatformVideoInput.Match Text    2. Microsoft Entra ID
+
+    # Select the Entra ID provider
+    PlatformHid.Type String    2
+    Builtin.Sleep    2
+
+Regenerate QR Code
+    # As long as we are in the login process, we can regenerate the QR code
+    PlatformHid.Keys Combo    Return
+    PlatformVideoInput.Match Text    https://microsoft.com/devicelogin    15
+
+
+Continue Log In With Remote User: Log In On Browser
+    [Arguments]    ${username}    ${domain}
+# Wait until the verification URL and login code are displayed
+    PlatformVideoInput.Match Text    https://microsoft.com/devicelogin
+    # Read the user code.
+    ${text} =    PlatformVideoInput.Read Text
+    ${user_code} =    StringUtils.FirstMatch    (https://)?microsoft.com/devicelogin\n((Login code: )?([A-Z0-9]+))    ${text}
+
+    Start Process    ${ENTRAID_COMMON}/BrowserWindow.py    ${username}@${domain}    %{E2E_PASSWORD}    ${user_code}    alias=RemoteAuth
```

Comment:
Do we want to run it with `RUNNING_OFFSCREEN` or we prefer to see the UI to record it, just in case?

Reply:
Sometimes it bugs out and types the password in the incorrect field (I couldn't understand why exactly), but it's something that we definitely do not want to record

Reply:
Let's keep this open, and see if I can handle it :)